src/api/search.api.js

```diff
@@ -272,6 +281,141 @@ async function searchLayerFeatures(outputProjection, queryString, layer, lang, c
+ *
+ * @param feature
+ * @param {String} queryString
+ * @param {String} searchFields
+ * @returns {Boolean}
+ */
+function isQueryInFeature(feature, queryString, searchFields) {
+    const queryStringClean = queryString
+        .trim()
+        .toLowerCase()
+        .normalize('NFD')
+        .replace(/[\u0300-\u036f]/g, '')
+    return searchFields.some((field) => {
+        const value = feature.values_[field]?.toString()
+        return !!value && value.trim().toLowerCase().includes(queryStringClean)
+    })
+}
+
+/**
+ * Searches for the query string in the KML layer
+ *
+ * @param {CoordinateSystem} outputProjection
+ * @param {String} queryString
+ * @param layer
+ * @returns {SearchResult[]}
+ */
+async function searchKmlLayerFeatures(outputProjection, queryString, layer) {
+    try {
+        const searchFields = ['name', 'description', 'id']
+        const features = parseKml(layer, outputProjection, [])
+        if (!features || !features.length) {
+            return []
+        }
+
+        const includedFeatures = features.filter((feature) =>
+            isQueryInFeature(feature, queryString, searchFields)
+        )
+        if (!includedFeatures.length) {
+            return []
+        }
+
+        return includedFeatures.map((feature) =>
+            createSearchResultFromLayer(layer, feature, outputProjection)
+        )
+    } catch (error) {
+        log.error(
+            `Failed to search layer features for layer ${layer.id}, fallback to empty result`,
+            error
+        )
+        return []
+    }
+}
+
+/**
+ * Searches for the query string in the GPX layer
+ *
+ * @param {CoordinateSystem} outputProjection
+ * @param {String} queryString
+ * @param layer
+ * @returns {SearchResult[]}
+ */
+async function searchGpxLayerFeatures(outputProjection, queryString, layer) {
+    try {
+        const searchFields = ['name', 'description', 'id']
+        const features = parseGpx(layer.gpxData, outputProjection, [])
+        if (!features || !features.length) {
+            return []
+        }
+        const includedFeatures = features.filter((feature) =>
+            isQueryInFeature(feature, queryString, searchFields)
+        )
+        if (!includedFeatures.length) {
+            return []
+        }
+        return includedFeatures.map((feature) =>
+            createSearchResultFromLayer(layer, feature, outputProjection)
+        )
+    } catch (error) {
+        log.error(
+            `Failed to search layer features for layer ${layer.id}, fallback to empty result`,
+            error
+        )
+        return []
+    }
+}
+
+/**
+ * Creates the SearchResult for a layer
+ *
+ * @param layer
+ * @param feature
+ * @param {ol/extent|null} extent
+ * @param {CoordinateSystem} outputProjection
+ * @returns {SearchResult}
+ */
+function createSearchResultFromLayer(layer, feature, outputProjection) {
+    const featureName = feature.values_.name || layer.name || ''
```

Comment:
you should prefer the use of `??` for variable declaration instead of `||` (which could lead to funny thing with boolean type, `??` is specifically made for "assign if value before is invalid")

Reply:
if i use ?? but the `feature.values_.name = ''` then it will take the empty string, but in this case i want that it does not take it but instead uses layer.name so i intentionally use `||`
i can add a comment to make it clear